modules/programs/urxvt.nix

```diff
@@ -0,0 +1,150 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.urxvt;
+
+in
+
+{
+  options.programs.urxvt = {
+    enable = mkEnableOption "rxvt-unicode terminal emulator";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.rxvt_unicode;
+      defaultText = "pkgs.rxvt_unicode";
+      description = "rxvt-unicode package to install.";
+    };
+
+    fonts = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = "List of fonts to be used.";
+      example = [ "xft:Droid Sans Mono Nerd Font:size=9" ];
+    };
+
+    keybindings = mkOption {
+      type = types.attrsOf types.string;
+      default = {};
+      description = "Mapping of keybindings to actions";
+      example = {
+        "Shift-Control-C" = "eval:selection_to_clipboard";
+        "Shift-Control-V" = "eval:paste_clipboard";
+      };
+    };
+
+    iso14755 = mkOption {
+      type = types.bool;
+      default = true;
+      description = "ISO14755 support for viewing and entering unicode characters.";
+    };
+
+    scroll = {
+      bar = mkOption {
+        type = types.submodule {
+          options = {
+            enable = mkOption {
+              type = types.bool;
+              default = true;
+              description = "Whether to enable the scrollbar";
+            };
+
+            style = mkOption {
+              type = types.enum [ "rxvt" "plain" "next" "xterm" ];
+              default = "plain";
+              description = "Scrollbar style.";
+            };
+
+            align = mkOption {
+              type = types.enum [ "top" "bottom" "center" ];
+              default = "center";
+              description = "Scrollbar alignment.";
+            };
+
+            position = mkOption {
+              type = types.enum [ "left" "right" ];
+              default = "right";
+              description = "Scrollbar position.";
+            };
+
+            floating = mkOption {
+              type = types.bool;
+              default = true;
+              description = "Whether to display an rxvt scrollbar without a trough.";
+            };
+          };
+        };
+        description = "Scrollbar settings.";
+      };
+
+      lines = mkOption {
+        type = types.int;
+        default = 64;
+        description = "Number of lines to save in the scrollback buffer.";
+      };
+
+      keepPosition = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to keep a scroll position when tty receives new lines.";
+      };
+
+      downOnInput = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to scroll to bottom on keboard input.";
+      };
+
+      downOnOutput = mkOption {
```

Comment:
I'd suggest `scrollOnOutput` to match `programs.gnome-terminal.profile.<name>.scrollOnOutput` and `programs.termite.scrollOnOutput`.

Yeah, it's a bit unfortunate to have `scroll.scrollOnOutput` but perhaps nice to have consistent naming here 🙂

Reply:
Great suggestion. Agree to both `scrollOnOutput` and `scrollOnKeystroke`, nice to be consistent.